src/interface/guide/components/BulletGraph.tsx

```diff
@@ -0,0 +1,203 @@
+  maximum: number;
+  /** Label to display on the main bar (e.g., "15 casts") */
+  actualLabel: string;
+  /** Label to display at the target marker (e.g., "Max 20") */
+  maximumLabel: string;
+  /** Color for the main progress bar */
+  barColor: string;
+  /** Performance ranges to display in the background */
+  performanceRanges?: PerformanceRange[];
+  /** Optional secondary metric (e.g., wasted time) */
+  secondaryMetric?: {
+    /** Value as percentage of fight duration (0-100) */
+    value: number;
+    /** Label to display (e.g., "5s capped") */
+    label: string;
+    /** Color for the secondary bar */
+    color: string;
+  };
+  /** Optional CSS class name */
+  className?: string;
+}
+
+/**
+ * BulletGraph component for visualizing performance metrics.
+ *
+ * @param actual - The actual value to display
+ * @param maximum - The maximum/target value (used for percentage calculation)
+ * @param actualLabel - Label to display on the main bar (e.g., "15 casts")
+ * @param maximumLabel - Label to display at the target marker (e.g., "Max 20")
+ * @param barColor - Color for the main progress bar
+ * @param performanceRanges - Optional performance ranges for the background
+ * @param secondaryMetric - Optional secondary metric (e.g., wasted time)
+ * @param className - Optional CSS class name
+ */
+export default function BulletGraph({
+  actual,
+  maximum,
+  actualLabel,
+  maximumLabel,
+  barColor,
+  performanceRanges,
+  secondaryMetric,
+  className,
+}: BulletGraphProps) {
+  const percentage = Math.min((actual / maximum) * 100, 100);
+
+  return (
+    <Container className={className}>
+      <GraphBackground>
+        {performanceRanges?.map((range, index) => (
+          <PerformanceZone key={index} width={range.width} color={range.color} />
+        ))}
+      </GraphBackground>
+
+      <MainBar width={percentage} color={barColor}>
+        <BarLabel>{actualLabel}</BarLabel>
+      </MainBar>
+
+      <TargetMarker>
+        <TargetLine />
+        <TargetLabel>{maximumLabel}</TargetLabel>
+      </TargetMarker>
+
+      {secondaryMetric && (
+        <SecondaryBar>
+          <SecondaryFill width={secondaryMetric.value} color={secondaryMetric.color} />
+          <SecondaryLabel position={secondaryMetric.value}>{secondaryMetric.label}</SecondaryLabel>
+        </SecondaryBar>
+      )}
+    </Container>
+  );
+}
+
+const Container = styled.div`
+  width: 100%;
+  position: relative;
+  display: flex;
+  flex-direction: column;
+  gap: 4px;
+`;
+
+const GraphBackground = styled.div`
+  width: 100%;
+  height: 32px;
+  background: rgba(0, 0, 0, 0.4);
+  border-radius: 4px;
+  overflow: hidden;
+  display: flex;
+  position: relative;
+  box-shadow: inset 0 1px 3px rgba(0, 0, 0, 0.4);
+  border: 1px solid rgba(255, 255, 255, 0.05);
+`;
+
+const PerformanceZone = styled.div<{ width: number; color: string }>`
+  width: ${(props) => props.width}%;
+  height: 100%;
+  position: relative;
+`;
```

Comment:
this doesn't set the `background-color` attribute using `color`, so the `PerformanceZone`s don't actually work.

```suggestion
const PerformanceZone = styled.div<{ width: number; color: string }>`
  width: ${(props) => props.width}%;
  background-color: ${(props) => props.color};
  height: 100%;
  position: relative;
`;
```